docs/pallas/gpu/reference.md

```diff
@@ -173,9 +173,119 @@ access pattern:
 Note that even though the number of active blocks hasn't changed, the total footprint of the data they
 access has halved! We get a much higher chance of getting L2 hits now.
 
-## Array layouts and reference transforms
+## Array layouts and memory reference transforms
 
-TODO
+In Pallas, the data structures you work with (arrays and references) have a
+**logical shape** (e.g., a 128x128 matrix). This
+logical shape must be mapped to a **physical representation** (how the data is
+actually represented in the GPU's memory). The specific mapping depends on where the
+data resides:
+
+1.  **Array Layouts:** Arrays are stored in register memory and we call this mapping
+    a _layout_. Layouts define how the elements of an array are
+    distributed across the registers available to the CUDA lanes that form a Pallas thread.
```

Comment:
Where does TMEM fit here? Maybe we can omit it from the docs for now, but I think it is also modelled using layouts?

Reply:
Yeah TMEM is its own special snowflake... We have yet another way to model its layouts, but they are a bit janky and not fully worked out yet.